src/clack-utils.ts

```diff
@@ -0,0 +1,346 @@
+      ),
+    );
+
+    return process.exit(1);
+  }
+
+  const loginUrl = new URL(
+    `${options.url}account/settings/wizard/${wizardHash}/`,
+  );
+
+  if (!hasSentryAccount) {
+    loginUrl.searchParams.set('signup', '1');
+    loginUrl.searchParams.set('project_platform', 'javascript-nextjs');
+  }
+
+  if (options.promoCode) {
+    loginUrl.searchParams.set('code', options.promoCode);
+  }
+
+  clack.log.info(
+    `${chalk.bold(
+      `Please open the following link in your browser to ${
+        hasSentryAccount ? 'log' : 'sign'
+      } into Sentry:`,
+    )}\n\n${chalk.cyan(loginUrl.toString())}`,
+  );
+
+  const loginSpinner = clack.spinner();
+
+  loginSpinner.start(
+    'Waiting for you to click the link above 👆. Take your time.',
+  );
+
+  const data = await new Promise<WizardProjectData>(resolve => {
+    const pollingInterval = setInterval(() => {
+      axios
+        .get<WizardProjectData>(`${options.url}api/0/wizard/${wizardHash}/`)
+        .then(result => {
+          resolve(result.data);
+          clearTimeout(timeout);
+          clearInterval(pollingInterval);
+          void axios.delete(`${options.url}api/0/wizard/${wizardHash}/`);
+        })
+        .catch(() => {
+          // noop - just try again
+        });
+    }, 500);
+
+    const timeout = setTimeout(() => {
+      clearInterval(pollingInterval);
+      loginSpinner.stop(
+        'Login timed out. No worries - it happens to the best of us.',
+      );
+      clack.outro(
+        'Please restart the Wizard and log in to complete the setup.',
+      );
+      return process.exit(0);
+    }, 180_000);
+  });
+
+  loginSpinner.stop('Login complete.');
+
+  return data;
+}
+
+export async function installPackage({
+  packageName,
+  alreadyInstalled,
+}: {
+  packageName: string;
+  alreadyInstalled: boolean;
+}): Promise<void> {
+  if (alreadyInstalled) {
+    const shouldUpdatePackage = await clack.confirm({
+      message: `The ${chalk.bold.cyan(
+        packageName,
+      )} package is already installed. Do you want to update it to the latest version?`,
+    });
+
+    abortIfCancelled(shouldUpdatePackage);
+    return;
+  }
+
+  const sdkInstallSpinner = clack.spinner();
+
+  let detectedPackageManager;
+  if (fs.existsSync(path.join(process.cwd(), 'yarn.lock'))) {
+    detectedPackageManager = 'yarn';
+  } else if (fs.existsSync(path.join(process.cwd(), 'package-lock.json'))) {
+    detectedPackageManager = 'npm';
+  } else if (fs.existsSync(path.join(process.cwd(), 'pnpm-lock.yaml'))) {
+    detectedPackageManager = 'pnpm';
+  } else {
+    detectedPackageManager = 'npm';
+  }
+
+  sdkInstallSpinner.start(
+    `${alreadyInstalled ? 'Updating' : 'Installing'} ${chalk.bold.cyan(
+      packageName,
+    )} with ${chalk.bold(detectedPackageManager)}.`,
+  );
+
+  try {
+    if (detectedPackageManager === 'yarn') {
+      await promisify(childProcess.exec)(`yarn add ${packageName}@latest`);
```

Comment:
Nit: This can run as execSync. Promisifying might be unnecessary.

Reply:
Using execSync here will block the event loop and prevents clack animations from showing.